index.js

```diff
@@ -703,135 +703,172 @@
 			});
 		}
 
+		/**
+		 *  Helper function to log amorphic errors.
+		 * @param {*} logger
+		 * @param {*} activity
+		 * @param {*} message
+		 * @param {*} logType
+		 * @param {*} logString
+		 */
+		function postCallErrorLog(logger, activity, message, logType, logString) {
+			let logBody = {
+				component: 'semotus',
+				module: 'processCall.failure',
+				data: {
+					call: remoteCall.name,
+					callTime: logTime(),
+					sequence: remoteCall.sequence
+				}
+			};
+
+			logBody.activity = activity;
+
+			if (logger.data) {
+				logBody.data.message = message;
+			}
+
+			logger[logType](logBody, logString);
+		}
+
+		/**
+		 * Helper function to identify if there's a postServerErrorHandler callback on the base controller
+		 * If there is, we execute the handler, and if we catch an error in the handler, we propogate it up to the logger.
+		 * @param {*} controller
+		 */
+		function resolveErrorHandler(logger, controller, type, remoteCall, remoteCallId, callContext, changeString) {
+			if (controller && controller['postServerErrorHandler']) {
```

Comment:
do we want to additionally check if what's in `controller['postServerErrorHandler']` is a function?

Reply:
Haha, that'll just be caught by the error handler, so I think we'll be alright. I'll create a type interface soon that people can utilize for this